cpp/src/parquet/page_index.h

```diff
@@ -26,9 +26,12 @@
 namespace parquet {
 
 class ColumnDescriptor;
```

Comment:
Can we perhaps use the relevant `type_fwd.h` includes instead of repeating all those fw-declarations?